### ISSUE TITLE
Build error on latest nightly: unknown feature array_value_iter_slice

### ISSUE THREAD

**Issue by kangalioo** [regexident/cargo-modules#70]
After updating to the latest nightly using `rustup update` and installing cargo-modules using `cargo +nightly install cargo-modules`, this build error occured:

```rust
error[E0635]: unknown feature `array_value_iter_slice`
  --> /home/kangalioo/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_arena-693.0.0/src/lib.rs:14:12
   |
14 | #![feature(array_value_iter_slice)]
   |            ^^^^^^^^^^^^^^^^^^^^^^
```

Presumably, the latest nightly has stabilized the array_value_iter_slice feature which the rustc-ap-rustc_arena crate is not prepared to deal with. Not sure what the best way to resolve this is on cargo-module's side.

The issue can be worked around by manually navigating into the faulty source file and commenting out the `#![feature(array_value_iter_slice)]` line.

**Comment by regexident** [comment]
Hey @kangalioo and thanks for the report!

This, among many other bugs, should be fixed with `0.5.0-beta.1`, which you can install via:

```bash
cargo install cargo-modules --version "0.5.0-beta.1"
```

**Comment by kangalioo** [comment]
Thank you! The new version immediately compiled successfully.

However, with the new version, I'm struggling to invoke it correctly. No matter what command I use after `cargo modules`, it always prints the same error message:
```rust
error: Found argument 'modules' which wasn't expected, or isn't valid in this context

USAGE:
    cargo-modules <SUBCOMMAND>

For more information try --help
```
This errors stays the same, no matter if I try `cargo modules --help`, `cargo modules graph`, or `cargo modules`. I also tried `cargo-modules`, but this command doesn't exist.

Is there something I can do to fix this argument parsing error?

**Comment by regexident** [comment]
Oh, I can vaguely remember a bug like this, which was fixed years ago. 🤔 
Looks like my recent major refactoring reintroduced it.

It basically boils down to `cargo` passing `modules` as an argument to `cargo-modules`, rather than dropping it:

`$ cargo-modules generate graph …`

```rust
std::env::args => ["/path/to/cargo-modules", "generate", "graph", …]
```

`$ cargo modules generate graph …`

```rust
std::env::args => ["/path/to/cargo-modules", "modules", "generate", "graph", …]
```

We basically have two options to fix this:

1. Pass a cleaned-up args vector (dropping the `"modules"`) to clap/structopt.

    Feels a bit hack-ish, but would work.

2. Add a `modules` umbrella root-command to structopt.

    This would probably mess with the output produced by `--help`, which is no good.

3. Add a positional hidden dummy argument.

    This is how it would be done in pure clap (we use structopt though):

    ```rust
    .arg(Arg::with_name("dummy")
        .hidden(true)
        .possible_value("modules"))
    ```

Option 3), the dummy argument would probably be the least invasive one.

**Comment by regexident** [comment]
Should be fixed with `0.5.0-beta.2` (I yanked `0.5.0-beta.1`).

**Comment by kangalioo** [comment]
It works, thank you!
